sound/soc/sof/ipc4-loader.c

```diff
@@ -178,6 +281,30 @@ struct sof_ipc4_fw_module *sof_ipc4_find_module_by_uuid(struct snd_sof_dev *sdev
 		}
 	}
 
+	/*
+	 * Do not attempt to load external library in case the maximum number of
+	 * firmware libraries have been already loaded
+	 */
+	if ((lib_id + 1) == ipc4_data->max_libs_count) {
+		dev_err(sdev->dev,
+			"%s: Maximum allowed number of libraries reached (%u)\n",
+			__func__, ipc4_data->max_libs_count);
+		return NULL;
+	}
+
+	/* The module can not be found, try to load it as a library */
+	ret = sof_ipc4_load_library_by_uuid(sdev, lib_id + 1, uuid);
+	if (ret)
+		return NULL;
+
+	/* Look for the module in the newly loaded library, it should be available now */
```

Comment:
the fw_lib is built by sof_ipc4_load_library_by_uuid, so we can return it by this function. Why we need to find it again ?

Reply:
The `sof_ipc4_find_module_by_uuid()` returns a pointer to the **fw module**, I think it would be awkward for the `sof_ipc4_load_library_by_uuid()`, which is loading a library (which can contain multiple modules) to return a pointer to a module also.
The very same loop must be done anyways as the manifest parsing is _not_ identifying the modules.

I think keeping the layers separate is better.